### ISSUE TITLE
Development dependencies not properly installed by the ci helper scripts

### ISSUE THREAD

**Issue by purbon** [elasticsearch/logstash#2725]
Hi,
  I found that this set of commands 

```
rake bootstrap
rake test:install-core
rake test:core
```

miss the development dependencies from logstash, they are installed but not available at runtime.

This is a regression bug, as this set of commands where working as expected since last Friday. Trying to scope when the problem was introduced.

Might be related with #2631?

Explanation for #2718.

**Comment by purbon** [comment]
Actually this issue started with the commit https://github.com/elasticsearch/logstash/commit/4edbf5d32bf532b27adcda1fa21c1f5d0ce0de28, do you have any idea what might happend @colinsurprenant ?

**Comment by purbon** [comment]
As I could debug today, even if we remove the ```.bundle``` and run ```bin/plugin install --development``` the config file generated is:

```
---
BUNDLE_PATH: /Users/purbon/work/logstash/vendor/bundle
BUNDLE_GEMFILE: /Users/purbon/work/logstash/Gemfile
BUNDLE_WITHOUT: development
BUNDLE_DISABLE_SHARED_GEMS: '1'
```
asking bundler to not install development dependencies.

**Comment by purbon** [comment]
@colinsurprenant @suyograo working on a fix for that, that avoid asking bundler to run in test environment without development dependencies.

**Comment by purbon** [comment]
Fixed in #2726
